src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -752,7 +754,8 @@ public void Fault(RequestChannel channel)
                     }
                     catch (OperationCanceledException)
                     {
-                        if (_timeoutHelper.CancellationToken.IsCancellationRequested)
+                        var cancelToken = _timeoutHelper.GetCancellationTokenAsync().Result;
```

Comment:
We should have a synchronous version. Calling .Result will wrap any exceptions in an aggregate exception so you need to use .GetAwaiter().GetResult() instead. That's too clumsy and specific to expect everybody using it to remember to do it correctly, so instead make the method TimeoutHelper.GetCancellationToken();